projects/fluid_simulation/shaders.h

```diff
@@ -0,0 +1,489 @@
+// Copyright 2017 Pavel Dobryakov
+// Copyright 2022 Google LLC
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+#ifndef __FLUID_SIMULATION_SHADERS_H
+#define __FLUID_SIMULATION_SHADERS_H
+
+#include "ppx/graphics_util.h"
+#include "ppx/grfx/grfx_buffer.h"
+#include "ppx/grfx/grfx_config.h"
+#include "ppx/grfx/grfx_image.h"
+#include "ppx/math_config.h"
+
+#include <iostream>
+#include <map>
+
+namespace FluidSim {
+
+class ProjApp;
+class FluidSimulation;
+
+// Pipeline interface, descriptor layout, sampler and uniform buffer used by compute shaders.
+struct ComputeResources
+{
+    ppx::grfx::PipelineInterfacePtr   mPipelineInterface;
+    ppx::grfx::SamplerPtr             mSampler;
+    ppx::grfx::DescriptorSetLayoutPtr mDescriptorSetLayout;
+};
+
+// Pipeline interface, descriptor layout, sampler and other resources used for graphics shaders.
+struct GraphicsResources
+{
+    ppx::grfx::PipelineInterfacePtr   mPipelineInterface;
+    ppx::grfx::VertexBinding          mVertexBinding;
+    ppx::grfx::DescriptorSetLayoutPtr mDescriptorSetLayout;
+    ppx::grfx::SamplerPtr             mSampler;
+    ppx::grfx::BufferPtr              mVertexBuffer;
+    ppx::grfx::BufferPtr              mTransformBuffer;
+};
+
+// Frame synchronization data.
+struct PerFrame
```

Comment:
I put it there because it deals with the graphics API directly.  `sim.h` only deals with the simulation aspects (yeah, it's not completely independent from the BW API, but it tries to be).